Several small improvements

bonjorno7 opened the pull request:
Added some comments from Momentum Mod's version of the shader to this one.
Improved consistency of comments by removing unnecessary periods.
Moved envmaplod caluclation to the dynamic state to prevent a crash that happened to be very common in Momentum Mod (not so much elsewhere, see the commit description for details).
Added defines for samplers in the cpp to make it easier to shuffle them around in the future.

bonjorno7 committed:
Improve comments

There's more and they're slightly more consistent

bonjorno7 committed:
Move envmaplod calculation to dynamic state

This prevents a crash that happens when switching from an LDR map without any PBR materials, to a HDR map with PBR materials on it.

bonjorno7 committed:
Add sampler defines

This makes it an awful lot easier to change things in the future

thexa4 commented on mp/src/materialsystem/stdshaders/pbr_dx9.cpp:
```diff
@@ -4,15 +4,33 @@
+#define SAMPLER_BASETEXTURE SHADER_SAMPLER0
+#define SAMPLER_NORMAL SHADER_SAMPLER1
+#define SAMPLER_ENVMAP SHADER_SAMPLER2
+#define SAMPLER_SHADOWDEPTH SHADER_SAMPLER4
```
Would you mind using consts here instead of defines?

Example: `const Sampler_t SAMPLER_BASETEXTURE = SHADER_SAMPLER0;`

bonjorno7 replied:
Sure

bonjorno7 replied:
Done

thexa4 reviewed (changes requested):
Thanks for the mipmap fix!

bonjorno7 committed:
Replace sampler defines with consts

Couldn't find where Sampler_t is definined, but it is used in a few places so I guess it exists.
